cashsvr_input_tof.c

```diff
@@ -101,79 +102,120 @@ int cash_tof_enable(bool enable)
 	return rc;
 }
 
-static int cash_tof_sys_init(bool high_accuracy, int devno, int plen)
+static int cash_tof_sys_init(bool high_accuracy, int devno, int plen,
+			     struct cash_tamisc_calib_params *calib_params)
 {
 	char* sns_mode_path;
-	int fd, rc;
-	struct passwd *pwd;
-	struct passwd *grp;
-	uid_t uid;
-	gid_t gid;
+	char* spad_calib_path;
+	char* um_offset_path;
+	char* buf;
+	int rc, cnt;
 
 	cash_tof_enable_path = (char*)calloc(plen + LEN_ENAB, sizeof(char));
 	if (cash_tof_enable_path == NULL) {
 		ALOGE("Memory exhausted. Cannot allocate.");
 		return -3;
 	}
 
-	pwd = getpwnam("system");
-	if (pwd == NULL) {
-		ALOGD("failed to get uid for system");
-		return 1;
-	}
-
-	uid = pwd->pw_uid;
-
-	grp = getpwnam("input");
-	if (grp == NULL) {
-		ALOGD("failed to get gid for input");
-		return 1;
-	}
-
-	gid = grp->pw_gid;
-
 	snprintf(cash_tof_enable_path, plen + LEN_ENAB,
 			"%s%d/enable_ps_sensor", sysfs_input_str, devno);
 
-	if (chown(cash_tof_enable_path, uid, gid) == -1) {
-		ALOGD("Cannot chown %s", cash_tof_enable_path);
-		return 1;
-	}
+	rc = cash_set_permissions(cash_tof_enable_path, "system", "input");
+	if (rc == -1)
+		return rc;
 
 	if (high_accuracy) {
 		sns_mode_path = (char*)calloc(plen + LEN_MODE, sizeof(char));
 		if (sns_mode_path == NULL) {
 			ALOGE("Memory exhausted. Cannot allocate.");
 			free(cash_tof_enable_path);
+			cash_tof_enable_path = NULL;
 			return -3;
 		}
 
 		snprintf(sns_mode_path, plen + LEN_MODE,
 			"%s%d/set_use_case", sysfs_input_str, devno);
 
-		if (chown(sns_mode_path, uid, gid) == -1) {
-			ALOGD("Cannot chown %s/enable_ps_sensor", sns_mode_path);
-			return 1;
-		}
-
-		fd = open(sns_mode_path, O_WRONLY);
-		if (fd < 0) {
-			ALOGD("Cannot open %s", sns_mode_path);
-			return 1;
+		rc = cash_set_permissions(sns_mode_path, "system", "input");
+		if (rc == -1) {
+			free(cash_tof_enable_path);
+			free(sns_mode_path);
+			cash_tof_enable_path = NULL;
+			return rc;
 		}
 
-		rc = write(fd, VL53L0_HIGH_ACCURACY, 1);
+		rc = cash_set_parameter(sns_mode_path, VL53L0_HIGH_ACCURACY,
```

Comment:
`sns_mode_path` not used beyond this line, should it be freed?

Reply:
Also, `cash_set_parameter` returns 0 on success, so check for `rc < 0` instead of `rc < 1`.